src/app/contexts/loadingContext.ts

```diff
@@ -0,0 +1,11 @@
+import { Dispatch, SetStateAction, createContext } from 'react';
+
+interface ILoadingContextValue {
+  isLoading: boolean;
+  setIsLoading?: Dispatch<SetStateAction<boolean>>;
```

Comment:
```suggestion
  setIsLoading: Dispatch<SetStateAction<boolean>>;
```
Тут setIsLoading всегда есть. Ты же задаешь ей дефолтное значение когда в контекст добавляешь. Можно убрать вопросик и избавиться везде от проверок, есть ли этот сеттер